CHANGELOG.md

```diff
@@ -2,6 +2,10 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+## main
+## Bug fixes 🐞
+* Fix memory leak in location component. ([#1093](https://github.com/mapbox/mapbox-maps-android/pull/1093))
```

Comment:
@Chaoba could you explain how was this reproduced?

Reply:
Open and close LocationComponentActivity several times could reproduce it.

Reply:
Yeah I mean - aren't we hiding the actual problem here? Why isn't unRegisterLocationConsumer called here? Shouldn't lifecycle call it always?

Reply:
Actually, we do call unRegisterLocationConsumer, but still have this issue.

Reply:
Hmm, how is this possible at all?

Reply:
It seems `locationEngine` still holds the reference even we call       `locationEngine.removeLocationUpdates(locationEngineCallback)
`

Reply:
Does this mean Android implementation is broken? Seems very suspicious, wouldn't it cause same issues with leaks in every app using location engine?

Reply:
This can only be reproduced with compass engine, guess locationEngine holds the reference of location plugin as well as sensorManager.

Reply:
Maybe we don't unsubscribe from compass engine somehow? I'd say compass is also very common component so if memory leaks were reproducible it would leak in many apps out there

Reply:
The root cause should be `locationEngine` holde the reference the callback during request location update even we remove it. And then location compass is kept and casue the leak.

Only request location update or have location compass can't reproduce this issue, must combile these two together.

Reply:
I've digged it a bit and it seems found the root cause. Thing is - we use Activity context to create SensorManager instance at [CompassEngine](https://github.com/mapbox/mapbox-maps-android/blob/9d5a59022b14d1ef8a1b30630d4fd5e703db5763/plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationCompassEngine.kt#L19). 

And callbacks, that we add / remove to the LocationEngine indeed are retained, even after remove is called. 

So the issue became visible with the CompassEngine addition, since it holds SensorManager, which holds Context, which holds (and is) Activity. 
It seems to me that we should use application context everywhere, so that in future such problems won't be possible at all. 

PS
Context for the issue is interesting as well. It seems that our implementation of LocationEngine in mapbox-android-core lacks certain logic cleaning the callback or otherwise such issues wouldn't be possible as well. 

The issue with LocationEngine keeping its listeners even after removeLocationUpdates has very long story, raised [here](https://github.com/mapbox/mapbox-events-android/issues/289), reworked [here](https://github.com/mapbox/mapbox-events-android/pull/290) and followed with the documented recommendation to avoid leaks using WeakReferences [here](https://github.com/mapbox/android-docs/issues/834) published [here](https://docs.mapbox.com/android/core/guides/#requesting-location-updates). 
I'll cut a ticket in the lib repo to apply the fix as well.

Reply:
https://github.com/mapbox/mapbox-events-android/issues/562